app/src/main/java/com/willowtree/vocable/room/CategoryWithPhrases.kt

```diff
@@ -1,13 +1,15 @@
 package com.willowtree.vocable.room
 
 import androidx.room.Embedded
+import androidx.room.Junction
 import androidx.room.Relation
 
 data class CategoryWithPhrases(
     @Embedded val category: Category,
     @Relation(
-        parentColumn = "identifier",
-        entityColumn = "category_id"
+        parentColumn = "category_id",
+        entityColumn = "phrase_id",
+        associateBy = Junction(CategoryPhraseCrossRef::class)
```

Comment:
So this class isn't an entity that's stored in the database. It's just a Room data class that helps with querying the junctions between categories and phrases. We made categories and phrases have a many-to-many relationship since a category has many phrases and a phrase can be part of many categories. There's more info [here](https://developer.android.com/training/data-storage/room/relationships#many-to-many).

Reply:
You can run queries on the `CategoryPhraseCrossRef`, but it'll just return single id values instead of objects. The `CategoryWithPhrases` will return full objects.

Reply:
Ah very cool, thanks for the explanation